ghost/core/core/server/web/parent/middleware/queue-request.js

```diff
@@ -0,0 +1,76 @@
+const debug = require('@tryghost/debug')('middleware:queue-request');
+const metrics = require('@tryghost/metrics');
+const path = require('node:path');
+const expressQueue = require('express-queue');
+
+const DEFAULT_QUEUE_LIMIT = 100;
+
+module.exports = function queueRequest(
+    config,
+    queueFactory = expressQueue,
+    logMetric = metrics.metric.bind(metrics)
+) {
+    config = {
+        ...{
+            limit: DEFAULT_QUEUE_LIMIT
+        },
+        ...config
+    };
+
+    debug('Initialising middleware with config:', config);
+
+    // @see https://github.com/alykoshin/express-queue#usage
+    const queue = queueFactory({
+        activeLimit: config.limit,
+        queuedLimit: -1 // Do not limit the number of queued requests
+    });
+
+    /**
+     * Available events:
+     * - queue - when a request is queued
+     * - dequeue - when a request is dequeued
+     * - process - when a request is being processed
+     * - reject - when a request is rejected
+     * - cancel - when a request is cancelled
+     * - complete - when a request has completed
+     *
+     * @see https://github.com/search?q=repo:alykoshin/mini-queue%20job._toState&type=code
+     */
+
+    queue.queue.on('queue', (job) => {
+        debug(`Request queued: ${job.data.req.path}`);
+
+        logMetric('request-queue', {
+            event: 'request-queued',
+            queueLength: job.queue.getLength()
+            // @todo: What else do we want to log here?
+        });
+    });
+
+    // @todo: This seems to fire even when a request is not queued
+    queue.queue.on('complete', (job) => {
+        debug(`Request completed: ${job.data.req.path}`);
+
+        logMetric('request-queue', {
+            event: 'request-completed',
+            queueLength: job.queue.getLength()
+            // @todo: What else do we want to log here?
+        });
+    });
+
+    return (req, res, next) => {
+        const ext = path.extname(req.path);
+        const staticAssetExts = ['.css', '.js', '.map', '.woff2', '.ico']; // @todo: extend this list to include other static asset types
```

Comment:
Nicely done here 🥇. We can hopefully scrape some existing logs for more data, or we can do some local testing to get more exts.

Reply:
Was actually thinking, would it be safe to do the inverse and if there is no extension automatically queue the request? I think we pretty much use extension-less urls everywhere 🤔

Reply:
Sounds like a plan. We can audit this in testing 👍.

Reply:
👍 Updated in https://github.com/TryGhost/Ghost/pull/19887/commits/fae6606bec8e59d2c2bc70470058970d968e368b